packages/flutter/test/material/card_theme_test.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' show Platform;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('CardTheme copyWith, ==, hashCode basics', () {
+    expect(const CardTheme(), const CardTheme().copyWith());
+    expect(const CardTheme().hashCode, const CardTheme().copyWith().hashCode);
+  });
+
+  testWidgets('Passing no CardTheme returns defaults', (WidgetTester tester) async {
+    await tester.pumpWidget(const MaterialApp(
+      home: Scaffold(body: Card()),
+    ));
+
+    final Container container = _getCardContainer(tester);
+    final Material material = _getCardMaterial(tester);
+
+    expect(material.clipBehavior, Clip.none);
+    expect(material.color, Colors.white);
+    expect(material.elevation, 1.0);
+    expect(container.margin, const EdgeInsets.all(4.0));
+    expect(material.shape, const RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(4.0)),
+    ));
+  });
+
+  testWidgets('Card uses values from CardTheme', (WidgetTester tester) async {
+    final CardTheme cardTheme = _cardTheme();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(cardTheme: cardTheme),
+      home: const Scaffold(body: Card()),
+    ));
+
+    final Container container = _getCardContainer(tester);
+    final Material material = _getCardMaterial(tester);
+
+    expect(material.clipBehavior, cardTheme.clipBehavior);
+    expect(material.color, cardTheme.color);
+    expect(material.elevation, cardTheme.elevation);
+    expect(container.margin, cardTheme.margin);
+    expect(material.shape, cardTheme.shape);
+  });
+
+  testWidgets('Card widget properties take priority over theme', (WidgetTester tester) async {
+    const Clip clip = Clip.hardEdge;
+    const Color color = Colors.orange;
+    const double elevation = 7.0;
+    const EdgeInsets margin = EdgeInsets.all(3.0);
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+
+    await tester.pumpWidget(MaterialApp(
+      theme: _themeData().copyWith(cardTheme: _cardTheme()),
+      home: const Scaffold(body: Card(
```

Comment:
The indentation could be slightly more conventional here and below.

```
home: const Scaffold(
  body: Card(
     ...
   ),
),
```

Reply:
done